dashboard/app/helpers/initial_section_creation_interstitial_helper.rb

```diff
@@ -3,10 +3,12 @@
 module InitialSectionCreationInterstitialHelper
   # Determine whether or not to show the initial section creation interstitial popup to a user
   # This interstitial should be shown only to teachers on their first sign-in
-  def self.show?(user)
+  def self.show?(user, request)
```

Comment:
I wonder if the second argument could be an optional boolean that defaults to false, rather than passing the entire request object? benefits would be, easier to test (though you already have that figured out), and extensibility in case hiding the modal could be useful in other situations in the future. 

totally just a suggestion though, this looks great!

Reply:
Good call, will do!